frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -1335,79 +1431,45 @@ export const multitabEditorLogic = kea<multitabEditorLogicType>([
+                        actions.editView(queryToOpen, view)
+                        tabAdded = true
                     }
-
-                    const queryToOpen = searchParams.open_query ? searchParams.open_query : view.query.query
-
-                    actions.editView(queryToOpen, view)
-                    tabAdded = true
-                    router.actions.replace(router.values.location.pathname)
                 } else if (searchParams.open_insight) {
                     if (searchParams.open_insight === 'new') {
                         // Add new blank tab
                         actions.createTab()
-                        tabAdded = true
                         router.actions.replace(router.values.location.pathname)
                         return
                     }
 
-                    // Open Insight
-                    const shortId = searchParams.open_insight
-                    const insight = await insightsApi.getByShortId(shortId, undefined, 'async')
-                    if (!insight) {
-                        lemonToast.error('Insight not found')
-                        return
-                    }
-
-                    let query = ''
-                    if (insight.query?.kind === NodeKind.DataVisualizationNode) {
-                        query = (insight.query as DataVisualizationNode).source.query
-                    }
-
-                    const queryToOpen = searchParams.open_query ? searchParams.open_query : query
-
-                    actions.editInsight(queryToOpen, insight)
-
-                    // Only run the query if the results aren't already cached locally and we're not using the open_query search param
-                    if (
-                        insight.query?.kind === NodeKind.DataVisualizationNode &&
-                        insight.query &&
-                        !searchParams.open_query
-                    ) {
-                        dataNodeLogic({
-                            key: values.dataLogicKey,
-                            query: (insight.query as DataVisualizationNode).source,
-                        }).mount()
-
-                        const response = dataNodeLogic({
-                            key: values.dataLogicKey,
-                            query: (insight.query as DataVisualizationNode).source,
-                        }).values.response
+                    try {
+                        // Load insight data first
+                        const shortId = searchParams.open_insight
+                        const insight = await insightsApi.getByShortId(shortId, undefined, 'async')
+                        if (!insight) {
+                            lemonToast.error('Insight not found')
+                            return
+                        }
 
-                        if (!response) {
-                            actions.runQuery()
+                        // Get the query from the insight
```

Comment:
Again, a lot of these comments just stink of AI - I'd appreciate if you could go through and clean all of this up